api/qan/collector.proto

```diff
@@ -2,56 +2,89 @@ syntax = "proto3";
+  rpc Collect(CollectRequest) returns (CollectResponse);
 }
 
-// AgentMessage contains several Metrics Bucket for one or more periods of time.
-message AgentMessage {
+message CollectRequest {
   repeated MetricsBucket metrics_bucket = 1;
 }
 
-// MetricsBucket is aggregated message created by qan-agent.
+// MetricsBucket is aggregated message created by pmm-agent.
 // Contains information about one query selected in defined way from query class in specific period of time.
 message MetricsBucket {
-  string queryid = 1; // md5 of digest_text/fingerprint.
-  string fingerprint = 2; // digest_text - query signature. Query without values.
+  // md5 of digest_text/fingerprint.
+  string queryid = 1;
+  // digest_text - query signature. Query without values.
+  string fingerprint = 2;
+  //
   // Dimension Group.
+  //
+
+  // Server.
   string d_server = 3;
-  string d_database = 4; // postgress database.
-  string d_schema = 5; // mysql database; postgress schema; mongo collection.
-  string d_username = 6; // client user name.
-  string d_client_host = 7; // client IP or hostname.
-  map<string, string> labels = 8; // Custom labels names:values.
-  string agent_uuid = 9; // Identifier of agent that collect and send metrics.
-  int64 period_start = 10; // Time when collection of bucket started.
-  uint32 period_length = 11; // Duration of bucket.
-  string metrics_source = 12; // Ex.: slowlog, perf schema, Mongo DB, Percona Server, etc.
-  string example = 13; // One of query example from set found in bucket.
+  // PostgreSQL database.
+  string d_database = 4;
+  // MySQL database; PostgreSQL schema; MongoDB collection.
+  // FIXME(askomorokhov) why collection?!
+  string d_schema = 5;
+  // client user name.
+  string d_username = 6;
+  // client IP or hostname.
+  string d_client_host = 7;
+  // Custom labels names:values.
+  map<string, string> labels = 8;
+  // Identifier of agent that collect and send metrics.
+  string agent_uuid = 9;
+  // Time when collection of bucket started.
+  // FIXME(askomorokhov) what unit? why not google.protobuf.Timestamp?
+  int64 period_start = 10;
+  // Duration of bucket.
+  // FIXME(askomorokhov) what unit? why not google.protobuf.Duration?
+  uint32 period_length = 11;
+  // Ex.: slowlog, perf schema, MongoDB, Percona Server, etc.
+  // FIXME(askomorokhov) Percona Server? why not enum?
+  string metrics_source = 12;
+  // One of query example from set found in bucket.
+  string example = 13;
   ExampleFormat example_format = 14;
-  uint32 is_truncated = 15; // Indicates if query examples is too long and was truncated.
+  // Indicates if query examples is too long and was truncated.
+  // FIXME(askomorokhov) why not boolean?
+  uint32 is_truncated = 15;
   ExampleType example_type = 16;
-  string example_metrics = 17; // JSON raw metrics for given query example.
+  // JSON raw metrics for given query example.
+  // FIXME(askomorokhov) wat?
+  string example_metrics = 17;
```

Comment:
@askomorokhov ^

Reply:
Please correct comment to `Metrics of given query example in JSON format.`

Reply:
I still don't get what it means.
Besides, we are sending a well structured message with strongly typed fields – and with any JSON object inside?

Reply:
This JSON object will simply displayed as table on UI below query examples without any aggregations.

Reply:
And what it will contain?

Reply:
all data from slow log entry for specific query.
Ex.:
```
# User@Host: mailboxer[mailboxer] @  [192.168.10.165]
# Thread_id: 11167745  Schema: board
# Query_time: 1.009400  Lock_time: 0.000190  Rows_sent: 4  Rows_examined: 1543719  Rows_affected: 0  Rows_read: 4
# Bytes_sent: 278  Tmp_tables: 0  Tmp_disk_tables: 0  Tmp_table_sizes: 0
# InnoDB_trx_id: 1500
# QC_Hit: No  Full_scan: Yes  Full_join: No  Tmp_table: No  Tmp_table_on_disk: No
# Filesort: No  Filesort_on_disk: No  Merge_passes: 0
#   InnoDB_IO_r_ops: 6415  InnoDB_IO_r_bytes: 105103360  InnoDB_IO_r_wait: 0.001279
#   InnoDB_rec_lock_wait: 0.000000  InnoDB_queue_wait: 0.000000
#   InnoDB_pages_distinct: 6430
SET timestamp=1346844943;
SELECT id,title,production_year FROM title WHERE title = 'Bambi';
```